sdk/core/core-http/test/mockHttp.ts

```diff
@@ -157,6 +157,13 @@ export class BrowserHttpMock implements HttpMockFacade {
   }
 
   timeout(method: HttpMethods, url: UrlFilter): void {
-    return this.mockHttpMethod(method, url, () => new Promise(() => {}));
+    return this.mockHttpMethod(
+      method,
+      url,
+      () =>
```

Comment:
alternative, have it return `Promise.resolve()` instead

Reply:
Seems like this works:

```ts
  timeout(method: HttpMethods, url: UrlFilter): void {
    return this.mockHttpMethod(
      method,
      url,
      () => Promise.resolve({})
    );
  }
```

Reply:
Perfect

Reply:
So, I couldn't do this. Seems like this really wants to have a new promise with no behavior returned. This `async () => {}` should work, but it clashes with the `MockResponse` type.

Soo... what I did was to change `MockResponseFunction` to return undefined optionally, `Promise<MockResponseData | void>`, also changed what used it to consider an optional value. I thought that would work.

Turns out the test was actually expecting this to fail! Why? 'cause the method here is `timeout`, so it makes sense that the test is expecting this to fail. Though it does this by throwing its own error if the request doesn't throw, and then checks that the error thrown is not the expected one. Weird!

To maintain this minimal, my fix near these lines was to set this last parameter to an async function that throws:

```ts
  timeout(method: HttpMethods, url: UrlFilter): void {
    return this.mockHttpMethod(method, url, async () => {
      throw new Error("Timeout");
    });
  }
```

Now it passes.